pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -87,11 +87,25 @@ func isExecutorTerminated(executorState v1beta1.ExecutorState) bool {
 	return executorState == v1beta1.ExecutorCompletedState || executorState == v1beta1.ExecutorFailedState
 }
 
-func driverPodPhaseToApplicationState(podPhase apiv1.PodPhase) v1beta1.ApplicationStateType {
-	switch podPhase {
+func driverStateToApplicationState(podStatus apiv1.PodStatus) v1beta1.ApplicationStateType {
+	switch podStatus.Phase {
 	case apiv1.PodPending:
 		return v1beta1.SubmittedState
 	case apiv1.PodRunning:
+		// TODO: upcoming Kubernenetes feature will make this code redundant
+		// https://github.com/kubernetes/enhancements/issues/753
+		for _, c := range podStatus.ContainerStatuses {
+			if (c.Name == config.SparkDriverContainerName) {
+				if c.State.Terminated != nil {
+					if (c.State.Terminated.ExitCode == 0) {
+						return v1beta1.SucceedingState
+					} else {
+						return v1beta1.FailingState
+					}
+				}
+				break;
```

Comment:
No need to add a semicolon at the end. Actually the `break` is not needed.

Reply:
I think this break is needed because I only want to look the driver container after I find driver container, I don't need to continue for loop.

Reply:
We always add sidecar containers after the main Spark container so you really only need to check the status of the first container if there's more than one container. I would change the for loop to the following:

```
if len(pod.Status.ContainerStatuses) > 1 {
  state := pod.Status.ContainerStatuses[0].State
  if state.Terminated == nil {
    return v1beta1.RunningState
  }
  if state.Terminated.ExitCode == 0 {
    return v1beta1.SucceedingState
  }
  return v1beta1.FailingState
}
```

Reply:
I see. I think I will keep the name-checking approach. since people can write their webhook that mess the container orders. this way feel safer. thanks